sources/yaml/src/main/java/io/smallrye/config/source/yaml/YamlConfigSource.java

```diff
@@ -0,0 +1,191 @@
+package io.smallrye.config.source.yaml;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.yaml.snakeyaml.Yaml;
+
+import io.smallrye.config.MapBackedConfigSource;
+
+/**
+ * Yaml config source
+ *
+ * @author <a href="mailto:phillip.kruger@redhat.com">Phillip Kruger</a>
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public class YamlConfigSource extends MapBackedConfigSource {
+
+    static final int ORDINAL = ConfigSource.DEFAULT_ORDINAL + 10;
+
+    private static final long serialVersionUID = -418186029484956531L;
+
+    public YamlConfigSource(String name, InputStream stream) throws IOException {
+        this(name, stream, ORDINAL);
+    }
+
+    public YamlConfigSource(String name, InputStream stream, int defaultOrdinal) throws IOException {
+        super(name, streamToMap(stream), defaultOrdinal, false);
+    }
+
+    public YamlConfigSource(String name, String str) {
+        this(name, str, ORDINAL);
+    }
+
+    public YamlConfigSource(String name, String str, int defaultOrdinal) {
+        super(name, stringToMap(str), defaultOrdinal, false);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static Map<String, String> streamToMap(InputStream inputStream) throws IOException {
+        final Map<String, Object> yamlInput;
+        try {
+            yamlInput = new Yaml().loadAs(inputStream, HashMap.class);
+            inputStream.close();
+        } catch (Throwable t) {
+            try {
+                inputStream.close();
+            } catch (Throwable t2) {
```

Comment:
I'm not familiar with this pattern, but should `t2` be used?

Reply:
It essentially replicates the exception handling used with `try`-with-resources, but in this case the secondary exception is suppressed, not the primary.

Reply:
But yes this was typo'd.